pkg/controllers/workapplier/apply.go

```diff
@@ -161,8 +162,11 @@ func (r *Reconciler) apply(
 			applyStrategy.ServerSideApplyConfig.ForceConflicts, isOptimisticLockEnabled, false,
 		)
 	default:
-		// An unexpected apply strategy has been set.
-		return nil, fmt.Errorf("unexpected apply strategy %s is found", applyStrategy.Type)
+		// An unexpected apply strategy has been set. Normally this will never run as the built-in
+		// validation would block invalid values.
+		wrappedErr := fmt.Errorf("unexpected apply strategy %s is found", applyStrategy.Type)
+		_ = controller.NewUnexpectedBehaviorError(wrappedErr)
+		return nil, wrappedErr
```

Comment:
I wonder why don't we return the unexpectedError here but return APIServer errors in other places?

Reply:
is that UnExpected erorr are "unExpected" so we don't want to record?

Reply:
Hi Ryan! It's still logged; the thing about this one is that the returned error will be formatted into a condition message which the user would be able to see, and the `unexpected behavior cannot be handled by the controller` part might read a bit weird; do we still want to expose them?

Reply:
I will make the changes if that is OK with you.

Reply:
I can see pros and cons on either solution